test/built-ins/String/prototype/search/cstm-search-invocation.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invocation of @@search property of user-supplied objects
+es6id: 21.1.3.15
+info: >
+    [...]
+    3. If regexp is neither undefined nor null, then
+       a. Let searcher be GetMethod(regexp, @@search).
+       b. ReturnIfAbrupt(searcher).
+       c. If searcher is not undefined, then
+          i. Return Call(searcher, regexp, «O»)
+features: [Symbol.search]
+---*/
+
+var regexp = {};
+var returnVal = {};
+var callCount = 0;
```

Comment:
That was my intent. I've included the assertion rather than remove the value, since it guards against the (however unlikely) case of repeated invocations of the method.